File: app/src/main/java/com/moviepass/responses/RestrictionsResponse.java
package com.moviepass.responses;

import com.moviepass.model.PopInfo;

/**
 * Created by anubis on 7/17/17.
 */

public class RestrictionsResponse {

    int countdown;
    String subscriptionStatus;
    boolean facebook;
    boolean has3d;
    boolean hasAllFormats;
    boolean proofOfPurchaseRequired;
    PopInfo popInfo;
    boolean hasActiveCard;

    public RestrictionsResponse() {
    }

    public int getCountdown() {
        return countdown;
    }

    public String getSubscriptionStatus() {
        return subscriptionStatus;
    }

    public boolean getFacebookPresent() {return facebook; }

    public boolean get3dEnabled() { return has3d; }

    public boolean getAllFormatsEnabled() { return hasAllFormats; }

    public boolean getProofOfPurchaseRequired() { return proofOfPurchaseRequired; }

    public PopInfo getPopInfo() {return popInfo; }

    public boolean getHasActiveCard() {return hasActiveCard; }

}

File: app/src/main/java/com/mobile/responses/RestrictionsResponse.kt
package com.mobile.responses

import com.google.gson.annotations.SerializedName
import com.mobile.model.*
import com.mobile.requests.PendingCharges


class RestrictionsResponse(
        var countDown: Int = 0,
        var subscriptionStatus: SubscriptionStatus = SubscriptionStatus.UNKNOWN,
        var facebook: Boolean = false,
        var has3d: Boolean = false,
        var hasAllFormats: Boolean = false,
        var proofOfPurchaseRequired: Boolean = false,
        var popInfo: PopInfo? = null,
        var hasActiveCard: Boolean = false,
        @SerializedName("capInfo")
        var cappedPlan:CappedPlan? = CappedPlan(),
        var alert: Alert? = null,
        var logoutInfo: LogoutInfo? = null,
        var blockRepeatShowings:Boolean = true,
        var subscriptionActivationRequired:Boolean = false,
        var canReactivate: CanReactivate? = null,
        var userSegments:List<Int> = emptyList(),
        var peakPassInfo:PeakPassInfo = PeakPassInfo(),
        val pendingChargesRestriction:PendingCharges?=null
)
